lib/common/json-schema-validator.js

```diff
@@ -0,0 +1,51 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = jsonSchemaValidatorFactory;
+
+jsonSchemaValidatorFactory.$provide = 'JsonSchemaValidator';
+jsonSchemaValidatorFactory.$inject = [
+    'ajv'
+];
+
+function jsonSchemaValidatorFactory(
+    Ajv
+) {
+    function JsonSchemaValidator () {
+        this._ajv = new Ajv({ allErrors: true,  verbose: true});
```

Comment:
Other schema validator may use special configuration, for example redfish-schema-validator doesn't need the "allErrors: true" as it doesn't need the special "context" validation.
So it's better to let the Validator supports additional options:
```
function JsonSchemaValidator(options) {
   this._ajv = new Ajv(options || {allErrors: true, verbose:  true});
}
```

Or we can completely remove the default option:
```
function JsonSchemaValidator(options) {
   this._ajv = new Ajv(options || {});
}
```